website/docs/examples/reactionroles.md

```diff
@@ -0,0 +1,383 @@
+
+```ts
+// This id should reflect your server id by now
+await bot.rest.upsertGuildApplicationCommands("1234", [...commands.values()]);
+```
+
+Also make sure to change the import at the top of the file.
+
+```ts
+// Old import. Delete this.
+import roles from "./src/commands/roles.js";
+// New import. Add this.
+import commands from "./src/commands/index.js";
+```
+
+Go ahead and start your bot, you will see the command is available on your server by typing `/roles reactions create`. If you try and execute the command it will fail since we have not yet added the handling of this command.
+
+## Command Execution Handling
+
+Let's make 2 very small files first. `src/events/index.ts` and `src/events/interactionCreate.ts`. Go to the interactionCreate file first.
+
+```ts
+import commands from "../commands/index.js";
+import { commandOptionsParser } from '@discordeno/utils';
+
+export const event: EventHandlers["interactionCreate"] = async function(interaction) {
+    if (interaction.type === InteractionTypes.ApplicationCommand) {
+        if (!interaction.data) return;
+
+        const command = commands.get(interaction.data.name);
+        if (!command) return;
+
+        await command.execute(interaction, commandOptionsParser(interaction));
+    }
+}
+```
+
+At this point, we are seeing an error from TypeScript, that the `command` does not have an `.execute()` handler. To add this we need to customize our command just a little bit. Let's make a interface for a custom Command object. Go to `src/commands/index.ts`
+
+```ts
+import { CreateApplicationCommand, Interaction } from '@discordeno/types';
+import roles from './roles.js';
+
+export const commands = new Map<string, CreateApplicationCommand>([
+    roles,
+].map((cmd) => [cmd.name, cmd]))
+
+export default commands;
+
+export interface Command extends CreateApplicationCommand {
+    /** Handler that will be executed when this command is triggered */
+    execute(interaction: Interaction, args: Record<string, any>): Promise<any>;
+}
+```
+
+Once this interface is made, we should edit the Map to use this as well.
+
+```ts
+export const commands = new Map<string, Command>([
+```
+
+Next we should edit the command file at `src/commands/role.ts` and edit it to have an execute handler.
+
+```ts
+import { CreateApplicationCommand, ApplicationCommandOptionTypes, ButtonStyles } from '@discordeno/types';
+
+const command: CreateApplicationCommand = {
+    name: "roles",
+    description: "Role management on your server.",
+    options: [
+        // Lot's of options here...
+    ],
+    async execute(interaction, args) {
+        
+    } 
+}
+```
+
+Once the execute handler is added, make sure to change the type to Command as well.
+
+```ts
+import { CreateApplicationCommand, ApplicationCommandOptionTypes, ButtonStyles } from '@discordeno/types';
+import { Command } from './index.js';
+
+const command: Command = {
+```
+
+Now that this is complete we should go 
```

Comment:
Is this abrupt stop intended?

Reply:
Haven't finished it, will finish it over time